pkg/daemon/ceph/client/config.go

```diff
@@ -270,3 +272,33 @@ func PopulateMonHostMembers(monitors map[string]*MonInfo) ([]string, []string) {
 
 	return monMembers, monHosts
 }
+
+// WriteCephConfig writes the ceph config so ceph commands can be executed
+func WriteCephConfig(context *clusterd.Context, clusterInfo *ClusterInfo) error {
```

Comment:
Any reason to not use CLI args to ceph commands like the operator? I recall doing a lot of work stripping out code that created config files unnecessarily.

Reply:
Related, this won't include user overrides to the ceph config.

Reply:
> Any reason to not use CLI args to ceph commands like the operator? I recall doing a lot of work stripping out code that created config files unnecessarily.

Actually all the rook code requires the ceph.conf to be generated. The operator, osd-prepare job, and osd-remove job all generate the config. It's just the containers that call `ceph` directly that use pure parameters instead of the ceph.conf.

> Related, this won't include user overrides to the ceph config.

The overrides are loaded a few calls deeper in the call chain in `mergeDefaultConfigWithRookConfigOverride()`

Reply:
Okay. I see that now. 👍